modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/Engine.scala

```diff
@@ -0,0 +1,27 @@
+package edu.gemini.seqexec.engine
+
+import scalaz._
+import Scalaz._
+
+/**
+  * A Map of `Sequence`s.
+  */
+case class Engine[+A](sequences: Map[Sequence.Id, Sequence[A]])
```

Comment:
I'm adding the conditions parameter to the Engine, but if you prefer we could put all the global values in a single Engine parameter (similar to `SequenceMetadata`). Is that what you have in mind?